base/io.jl

```diff
@@ -499,11 +499,15 @@ read!(filename::AbstractString, a) = open(io->read!(io, a), convert(String, file
     readuntil(stream::IO, delim; keep::Bool = false)
     readuntil(filename::AbstractString, delim; keep::Bool = false)
 
-Read a string from an I/O stream or a file, up to the given delimiter.
+Read a string from an I/O `stream` or a file, up to the given delimiter.
 The delimiter can be a `UInt8`, `AbstractChar`, string, or vector.
 Keyword argument `keep` controls whether the delimiter is included in the result.
 The text is assumed to be encoded in UTF-8.
 
+Returns a `String` if `delim` is an `AbstractChar` or a string
+or otherwise returns a `Vector{typeof(delim)}`.   See also [`copyuntil`](@ref)
+to instead write in-place to another stream (which can be a preallocated [`IOBuffer`](@ref)).
```

Comment:
Unless the `s` was intentional:
```suggestion
Return a `String` if `delim` is an `AbstractChar` or a string
or otherwise return a `Vector{typeof(delim)}`.   See also [`copyuntil`](@ref)
to instead write in-place to another stream (which can be a preallocated [`IOBuffer`](@ref)).
```

Reply:
The `s` was intentional — grammatically, I was thinking of the implicit subject being `copyuntil` (or "it").   i.e. "`copyuntil` returns a …".     It's descriptive, not imperative.

Reply:
Interestingly, the descriptive tense "Returns …" is used 45 times in the Julia docs, while the imperative tense "Return …" is used 288 times.

Reply:
See #50301 if you want to have a longer discussion.